### DISCUSSION TITLE
lerna watch compiles endlessly.

### DISCUSSION THREAD

**Discussion by john-twigg-ck** [lerna/lerna#3621] [Q&A]
I'm running
>lerna watch -- lerna run compile

And it keeps looping. Something in the output is re-triggering the build. But I'm finding no way to diagnose the issue. 
Does anyone have any experience with this? What ARE the inputs to `lerna watch` because its not clear.

PS: I don't have this problem with `nodemon` 
Example:
```
"start:watch": "nodemon -e js,ts,graphql --watch codegen.yml --watch src --watch ../schema/ --exec \"npm run codegen:graphql && npm run compile && npm run start\""
````

I have 7 projects, and the `nodemon` watch eventually settles down and doesn't loop.

**Comment by ghiscoding** [comment]
You're missing the scope which is probably why it loops endlessly, you need to scope to target which package to build. You can see that in their [Lerna Watching](https://lerna.js.org/docs/features/workspace-watching#examples) docs

```sh
lerna watch -- lerna run build --scope=\$LERNA_PACKAGE_NAME
```
Note that there's 2 sections in lerna watch, which is separated by ` -- `, the left side is for lerna watch itself, while the right side is for the targeted package execution

Also if you want to build not just that single package but also its dependents, then you can use (which is also displayed in the docs)
```sh
lerna watch -- lerna run build --scope=\$LERNA_PACKAGE_NAME --include-dependents
```
I haven't tried Lerna Watch itself, but I'm using this same approach with Lerna-Lite Watch and it works totally fine

**Comment by fahslaj** [comment]
Hi @john-twigg-ck , you could run into this issue if the command you are running (in this case, `lerna run compile`) is creating or editing files within the package directory that is being watched. `lerna watch` will ignore files specified in `.gitignore`, so as long as your built assets are in an ignored directory, this should not happen.
